robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents Report Portal in ReportPortal
+
+    This holds the properties and function to interact with the Report Portal
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
+
+    @property
+    def api_url(self):
+        """Super url of report portal to be used as a base in all child classes
+        :returns Base url for API request
+        """
+        _api_url = f'{self.rp_url}/api/v1/{self.rp_project}/'
```

Comment:
for a sake of future compatibility, I'd parametrize the API version (put it into properties file).
`.../api/{self.rp_api}/...`

Reply:
Can precisely explain ?

I don't think we should have an component wise API from properties file. We can tweak/append the per component in their own classes. As I am doing it here for Launch and Tests class.

Reply:
this is something that might be more easily addressed after robottelo's config is revamped. we can table this for now.

Reply:
Agreed @JacobCallahan, and again something we can iterate on with this design.